_posts/2018-09-18-artsy-frontend-history.markdown

```diff
@@ -0,0 +1,108 @@
+
+We moved our API from Rails to [Grape][], partially motivated by a desire to build an iOS application that would
+consume this API. We faced a lot of problems with the split, including SEO problems, severe page load times,
+maintaining duplicated backend and frontend UI templates, slow test suites, and poor developer productivity. We
+took the project of building our mobile web frontend, m.artsy.net (still known as "martsy" internally) as an
+opportunity to address these problems.
+
+We built our new site with [Node.js][node] since it easily allowed server-side rendering. We split out areas of
+concern into separate "apps", with their own bundled CSS/JS to help page load times. We server-side rendered
+above-the-fold content and used client-side JS to load the rest, which helped SEO and user experience. We took a
+[BEM][]-like approach to our CSS, which helped developer productivity. Our technical decisions were driven
+primarily by our desire to create great user experiences.
+
+And because we are an open source by default organization, we collected these approaches into an open source
+project called [Ezel][].
+
+We used Ezel for a few years without too much change in our web frontend stack. It proved really useful for helping
+build other web apps – CMS systems for our partners, auction-management systems for our admins, all kinds of
+projects – and most of those projects started on Heroku before moving to our Ops Works stack as needed. Our
+frontend mindset at the time (2015) was focused on getting to a stable, predictable stack. However... we started
+experimenting with React around the same time.
+
+CoffeeScript and Backbone were still working for us, and we still use them in production in many systems. However,
+the state of the art in web development moved on. When I joined the auctions team and helped maintain one of our
+CoffeeScript+Backbone apps, I was _very_ confused about how data flowed from one part of the app to another, across
+languages, with a lot of magic happening. I think that's typical in these kinds of apps – "convention over
+configuration" is a good mantra _if_ you can expect that incoming engineers are familiar with the conventions.
+That's just not the case anymore.
+
+By 2016, we launched our [first app built with React][auctions], which both proved the technology was ready for
+production use _and_ convinced our engineers that React is simply a better paradigm for building user interfaces.
+
+## React
+
+By 2017, the divisions between our mobile frontend and web frontend teams had been totally dissolved (as they
+should – the division between mobile and web developers is a false dichotomy). Our [2017 tech stack
+post][2017_review] discusses this in depth, but our goal was really to unify the paradigm that frontend engineers
+at Artsy use to build user interfaces, whether that's on mobile or web. React and React Native were our answer to
+that challenge.
+
+TODO: stitch, typescript, palette, reaction, etc
```

Comment:
Yeah, this sets up the start of 2017 well 👍

Reply:
SUSPENSE!

Reply:
Worth looking at https://artsy.github.io/blog/2017/09/05/Modernizing-Force/ and https://artsy.github.io/blog/2017/11/27/Babel-7-and-TypeScript/ since it covers some history

Reply:
@ashfurrow  Re Palette, it would be nice to introduce that in an entirely separate blog post and cover it exhaustively. There's a lot of technical decisions in there that need to be covered, from `styled-system` to universal components. We still have more work to do on this before we unveil it to the community. 

Also, we have a new isomorphic routing framework that has allowed us to build out all of our new apps. I would also like to introduce that alongside Palette but in a separate stand alone blog post. Since it involves Relay (which so often lacks information), it's important that we keep things visible.

Reply:
I think it's fine to mention something "we're exploring a design system, based on styled systems called palette" and keep it at that kinda level

I definitely want to see ^ though

Reply:
Sure, I get that desire for a separate post. I'm uneasy waiting on other blog posts, however, for two reasons:

- Blog posts are snapshots of time, and change is constant, so we'll never be in an optimum position to write about this stuff.
- I'm still waiting for a Reaction 1.0 launch blog post 😂 

I think alluding to Palette and isomorphic routing, but not going into details, might be a good compromise. What do you think?

Reply:
That sounds good :) I think some kind of teaser might be worthwhile (but without linking directly to palette) -- maybe a mention of styled-system would be sufficient and just a comment around universal relay rendering?